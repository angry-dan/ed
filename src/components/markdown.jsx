import React from "react";
import marked from "marked";
const ComponentMarkdown = ({body}) => <div className="markdown flow-text" dangerouslySetInnerHTML={{__html: marked(body)}}></div>;
class ComponentMarkdownForm extends React.Component {
  render () {
    return <textarea onChange={(event) => this.props.onChange({body: event.target.value})} value={this.props.body} className="materialize-textarea" placeholder="Enter text here in *markdown* _format_" />
  }
}
export default {
  'description': 'A markdown field',
  'component': ComponentMarkdown,
  'componentForm': ComponentMarkdownForm,
  'defaultState': {body: 'We can write in **Bold**, *italic*.'}
};
