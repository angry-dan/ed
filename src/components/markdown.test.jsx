import React from 'react';
import {ComponentMarkdown, ComponentMarkdownForm} from "./markdown";
import renderer from 'react-test-renderer';

const testBody = `Some markdown
**which includes bold** _italic_ and 
- a list
- of things`;

test('component renders correctly', () => {
  const tree = renderer.create(<ComponentMarkdown body={testBody} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('component form renders correctly', () => {
  const tree = renderer.create(<ComponentMarkdownForm body={testBody} />).toJSON();
  expect(tree).toMatchSnapshot();
})
