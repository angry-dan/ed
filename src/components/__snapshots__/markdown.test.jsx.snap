// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component form renders correctly 1`] = `
<textarea
  className="materialize-textarea"
  onChange={[Function]}
  placeholder="Enter text here in *markdown* _format_"
  value="Some markdown
**which includes bold** _italic_ and 
- a list
- of things"
/>
`;

exports[`component renders correctly 1`] = `
<div
  className="markdown flow-text"
  dangerouslySetInnerHTML={
    Object {
      "__html": "<p>Some markdown
    <strong>which includes bold</strong> <em>italic</em> and </p>
    <ul>
    <li>a list</li>
    <li>of things</li>
    </ul>
    ",
    }
  }
/>
`;
